Use the specified version of nginx.

Don't store or look for stuff in $HOME, use the passenger directory.

Index: src/ruby_supportlib/phusion_passenger.rb
--- src/ruby_supportlib/phusion_passenger.rb.orig
+++ src/ruby_supportlib/phusion_passenger.rb
@@ -34,7 +34,7 @@ module PhusionPassenger
   VERSION_STRING = '6.0.4'
 
   # Tip: find the SHA-256 with ./dev/nginx_version_sha2 <VERSION>
-  PREFERRED_NGINX_VERSION = '1.17.3'
+  PREFERRED_NGINX_VERSION = '${NGINX_VERSION}'
   NGINX_SHA256_CHECKSUM = '3b84fe1c2cf9ca22fde370e486a9ab16b6427df1b6ea62cdb61978c9f34d0f3c'
 
   # Packaging may be locked to an older version due to the specific module configuration being
@@ -323,18 +323,6 @@ private
   # example `sudo` preserves it. That's why we don't respect it by
   # default.
   def self.home_dir(respect_home_env = false)
-    if respect_home_env
-      home = ENV['HOME'].to_s
-    end
-    if home.nil? || home.empty?
-      require 'etc' if !defined?(Etc)
-      begin
-        home = Etc.getpwuid(Process.uid).dir
-      rescue ArgumentError
-        # Unknown user.
-        home = ENV['HOME']
-      end
-    end
-    return home
+    File.dirname(File.dirname(File.expand_path(__FILE__)))
   end
 end if !defined?(PhusionPassenger::VERSION_STRING)
